new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItInitContainers.java

```diff
@@ -0,0 +1,443 @@
+
+    createSecretWithUsernamePassword(adminSecretName, domainNamespace, "weblogic", "welcome1");
+
+    // create encryption secret
+    logger.info("Creating encryption secret");
+
+    createSecretWithUsernamePassword(encryptionSecretName, domainNamespace, "weblogicenc", "weblogicenc");
+  }
+
+
+  @AfterAll
+  public void tearDownAll() {
+    //delete operator
+    uninstallOperator(opHelmParams);
+  }
+
+  /**
+   * Add initContainers at domain spec level and verify the admin server pod executes initContainer command.
+   * Test fails if domain crd can't add the initContainers or
+   * WebLogic server pod doesn't go through initialization and ready state
+   */
+  @Test
+  @DisplayName("Add initContainers at domain spec level and verify the admin server pod executes initContainer command "
+      + " and starts the admin server pod")
+  public void testDomainInitContainer() {
+    logger.info("Installing and verifying domain");
+    assertTrue(createVerifyDomain(domain1Namespace, domain1Uid, "spec"),
+        "can't start or verify domain in namespace " + domain1Namespace);
+
+    //check if init container got executed
+    assertTrue(assertDoesNotThrow(() -> getPodLog(domain1Uid + "-admin-server", domain1Namespace,"busybox")
+            .contains("Hi from Domain"),
+        "failed to init busybox container command for admin server"));
+    assertTrue(assertDoesNotThrow(() -> getPodLog(domain1Uid + "-managed-server1", domain1Namespace,"busybox")
+            .contains("Hi from Domain"),
+        "failed to init busybox container command for managed server1"));
+    assertTrue(assertDoesNotThrow(() -> getPodLog(domain1Uid + "-managed-server2", domain1Namespace,"busybox")
+            .contains("Hi from Domain"),
+        "failed to init busybox container command for managed server2"));
+
+  }
+
+  /**
+   * Add initContainers to adminServer and verify the admin server pod executes initContainer command
+   * and starts the admin server pod.
+   * Test fails if domain crd can't add the initContainers or
+   * weblogic server pod doesn't go through initialization and ready state
+   */
+  @Test
+  @DisplayName("Add initContainers to adminServer and verify the admin server pod executes initContainer command ")
+  public void testAdminServerInitContainer() {
+    assertTrue(createVerifyDomain(domain2Namespace, domain2Uid, "adminServer"),
+        "can't start or verify domain in namespace " + domain2Namespace);
+
+    //check if init container got executed for admin server pod
+    assertTrue(assertDoesNotThrow(() -> getPodLog(domain2Uid + "-admin-server", domain2Namespace,"busybox")
+            .contains("Hi from AdminServer"),
+        "failed to init busybox container command for admin server"));
+  }
+
+  /**
+   * Add initContainers to adminServer and verify the managed server pods in cluster execute initContainer command
+   * before starting the admin server pod.
+   * Test fails if it cannot include the initContainers or
```

Comment:
Test fails if it cannot include the initContainers for clustered members or some other pods?

Reply:
done